### PR TITLE
Support newer Verilator in backwards compatible way

### PR DESCRIPTION
**Pull request by ktbarrett**
xref #3896.

### CODE REVIEW

File: cocotb/share/lib/vpi/VpiCbHdl.cpp [hunk 1]
```diff
@@ -379,6 +379,9 @@ int VpiSignalObjHdl::set_signal_value(s_vpi_value value_s,
 
     switch (action) {
         case GPI_DEPOSIT:
+#ifdef VERILATOR
```
Comment (marlonjames) [inline cocotb/share/lib/vpi/VpiCbHdl.cpp:382]: I think a comment here would be good, explaining why it's needed, even if 2.0 will remove it...

**Comment by ktbarrett** [comment]
Strange it isn't covering the new line, cocotb_build_libs.py says the define is VERILATOR?

**Comment by marlonjames** [comment]
Verilator CI isn't being run. You can see it's missing from the output of the generate script.

**Comment by ktbarrett** [comment]
@marlonjames Do we trust to merge this as is, or should we wait until Verilator CI changes are backported?

**Review by marlonjames — APPROVED** [review]
I'm okay with this, assuming we don't want to wait and bump the minimum Verilator version for 1.9.

**Review by cmarqu — APPROVED** [review]

**Pull request merged**